rust/lance/src/dataset/cleanup.rs

```diff
@@ -193,50 +210,92 @@ impl<'a> CleanupTask<'a> {
         Ok(())
     }
 
-    async fn delete_unreferenced_files(&self) -> Result<RemovalStats> {
+    async fn delete_unreferenced_files(
+        &self,
+        inspection: CleanupInspection,
+    ) -> Result<RemovalStats> {
         let removal_stats = Mutex::new(RemovalStats::default());
+        let verification_threshold = utc_now() - Duration::days(UNVERIFIED_THRESHOLD_DAYS);
         let unreferenced_paths = self
             .dataset
             .object_store
             .read_dir_all(&self.dataset.base, Some(self.before))
             .await?
-            .try_filter_map(|path| future::ready(self.path_if_not_referenced(path, &removal_stats)))
+            .try_filter_map(|obj_meta| {
+                // If a file is new-ish then it might be part of an ongoing operation and so we only
+                // delete it if we can verify it is part of an old version.
+                let maybe_in_progress =
+                    !self.delete_unverified && obj_meta.last_modified >= verification_threshold;
+                let path_to_remove =
+                    self.path_if_not_referenced(obj_meta.location, maybe_in_progress, &inspection);
+                if path_to_remove.is_ok() && path_to_remove.as_ref().unwrap().is_some() {
+                    removal_stats.lock().unwrap().bytes_removed += obj_meta.size as u64;
+                }
+                future::ready(path_to_remove)
+            })
             .boxed();
 
-        let old_manifests = self.old_manifests.lock().unwrap().clone();
+        let old_manifests = inspection.old_manifests.clone();
         let num_old_manifests = old_manifests.len();
-        let old_manifests_stream = stream::iter(old_manifests).map(Result::<Path>::Ok).boxed();
 
+        let manifest_bytes_removed = stream::iter(&old_manifests)
+            .then(|path| async move { self.dataset.object_store.size(path).await })
+            // .buffer_unordered(num_cpus::get())
```

Comment:
Can we restore this? Might be require a call to `boxed()` in the chain if the higher-order lifetime issue comes up here.

Reply:
I tried with no luck.  No amount of inserting `boxed` or `async move` has made any difference.

Reply:
I did add a comment explaining why it is commented out